Updating to purescript 0.7.0

SimonRichardson opened the pull request:
Tests are currently failing, need to investiagte why.

Currently :
- Have to jump through hoops because of `Data.List` vs `Data.Array`. It
   might be wiser to just move to `Data.List` and be done with it.

SimonRichardson committed:
Updating to purescript 0.7.0

Tests are currently failing, need to investiagte why.

Currently :

 - Have to jump through hoops because of `Data.List` vs `Data.Array`. It
 	 might be wiser to just move to `Data.List` and be done with it.

jdegoes commented:
@garyb Can you review / merge / cut a new release?

@SimonRichardson Thanks!

garyb commented on bower.json:
```diff
@@ -17,10 +17,12 @@
-    "purescript-integers": "~0.1.0",
-    "purescript-globals": "~0.1.6"
+    "purescript-string-parsers":  "^0.5.0",
+    "purescript-pathy":           "SimonRichardson/purescript-pathy#meltdown",
```
Please can you update this now there's a new `pathy` release?

SimonRichardson committed:
Updated to the latest release of purescript-pathy

Print port as a number
 - Prevent forcing the printing of the port to a number

SimonRichardson commented:
> Tests are currently failing, need to investigate why.

@garyb I need to investigate why this is happening, might get time this weekend, but unsure!

garyb commented:
Ok, cool! I can't see anything obviously wrong in the updates you made. If I get around to it before you and figure it out I'll merge this anyway and then apply the fixes.

SimonRichardson commented:
I've fixed one of the issues, the fact that the port was rendered as a number and not a int (easy fix), but I'm currently working out why we fail when running the parser, even though it looks correct.

garyb commented:
Which case(s) is it failing on?

SimonRichardson commented:
So this is the output from one of them.

```
Trying to parse telnet://192.0.2.16:80/
      printURI: telnet://192.0.2.16:80/
          show: Left (URI (Just (URIScheme "telnet")) (HierarchicalPart (Just (Authority (Nothing) [Tuple (IPv4Address "192.0.2.16") (Just (80))])) (Just (Right (rootDir)))) (Nothing) (Nothing))
```

garyb commented:
Odd, that should be passing I think... I'll check it out and have a look too.

garyb commented:
Ah, that one did pass, some of the others are failing though.

garyb commented:
Oh, actually, I think these tests always failed: they're unimplemented features (email addresses, relative URLs, news/tel/urn urls that I haven't figured out which rule is being broken).

Can you group the failing tests together and `log` "Failing tests" or something before them please? That'll do for now, unless you really feel like fixing the parser for the less usual cases. :smile:

SimonRichardson committed:
Combine all the failing test cases

Currently the URI package doesn't pass all the tests, so I've grouped
them together to make it easier to understand what's failing.

SimonRichardson commented:
@garyb I think it's best to release this as is, and I'll have a play at updating the parser later on.

garyb commented:
Great! Thanks for working on this.

SimonRichardson commented:
sure nps.

garyb commented:
It appears this wasn't released at all before, so I've released it as `v0.1.0-rc.1` for now. I'll try to fix up the parser issues before making a "proper" release.
